app/models/course.rb

```diff
@@ -163,6 +171,19 @@ def import_students(file, header_map, header_row_exists)
     end
   end
 
+  def create_user(github_username, email)
+    github_user = github_user_lookup(github_username)
+    return nil unless github_user 
+    User.create(
+          provider: ENV['OMNIAUTH_STRATEGY'],
+          uid: github_user.data.user.databaseId,
+          email: email,
+          name: github_user.data.user.name,
+          username: github_user.data.user.login,
+          password: Devise.friendly_token[0,20],
```

Comment:
Is this automatically creating a password for the user that is created? Is that not a problem because the user can simply log in via OAuth? If so, then do we _need_ a password?

Reply:
I agree, I would test if this causes problems when a user logs in for the first time after their User record was created with this method.

Reply:
So it turns out that the code for adding in a password was already in the code base.  We do this for users that are created when they login with GitHub via OAuth for the first time also.

My vague recollection from four years ago is that Devise requires that you have a password even if it is never used.    That's why we generate a random 20 character string.  The password is never used; it's just to satisfy Devise.